spt/features/playerio.cpp

```diff
@@ -283,25 +284,14 @@ int __fastcall PlayerIOFeature::HOOKED_GetButtonBits_Func(void* thisptr, int edx
 
 	if (bResetState == 1)
 	{
-		static bool duckPressed = false;
-
-		if (duckspam)
-		{
-			if (duckPressed)
-				duckPressed = false;
-			else
-			{
-				duckPressed = true;
-				rv |= (1 << 2); // IN_DUCK
-			}
-		}
-		else
-			duckPressed = false;
+		static int keyPressed = 0;
+		keyPressed ^= spamButtons;
+		rv |= keyPressed & spamButtons;
```

Comment:
If I read this correctly this has kinda weird behavior, when you unpress spam for a certain key you have one of two cases: either the flag is set in keyPressed or it's not. This state is maintained until the spam is activated again. If the flag is set, then when the spam is next activated the key will be pressed on the second tick. If the flag is not set, then the key will be pressed on the first tick.

I think this would work the same every time:
```
static int keyPressed = 0;  
keyPressed = (keyPressed ^ spamButtons) & spamButtons;  
rv |= keyPressed;  
```